service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/PractitionerAgentPersonMapper.java

```diff
@@ -0,0 +1,78 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+import java.util.Optional;
+
+import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.Practitioner;
+import org.hl7.fhir.dstu3.model.PractitionerRole;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.github.mustachejava.Mustache;
+
+import lombok.RequiredArgsConstructor;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.PractitionerAgentPersonMapperParameters;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Component
+public class PractitionerAgentPersonMapper {
+
+    private static final Mustache AGENT_STATEMENT_TEMPLATE = TemplateUtils
+        .loadTemplate("ehr_agent_person_template.mustache");
+    private final MessageContext messageContext;
+    private final OrganizationToAgentMapper organizationToAgentMapper;
+
+    public String mapPractitionerToAgentPerson(Practitioner practitioner, Optional<PractitionerRole> practitionerRole,
+        Optional<Organization> organization) {
+        var builder = PractitionerAgentPersonMapperParameters.builder()
+            .practitionerId(messageContext.getIdMapper().getOrNew(ResourceType.Practitioner, practitioner.getIdElement().getIdPart()));
+
+        buildPractitionerPrefix(practitioner).ifPresent(builder::practitionerPrefix);
+        buildPractitionerGivenName(practitioner).ifPresent(builder::practitionerGivenName);
+        buildPractitionerFamilyName(practitioner).ifPresent(builder::practitionerFamilyName);
+
+        practitionerRole.flatMap(this::buildPractitionerRole).ifPresent(builder::practitionerRole);
+        organization.ifPresent(organization1 -> {
```

Comment:
the same result should give
`
organization.map(organizationToAgentMapper::mapOrganizationToAgentInner).map(builder::organization);
`
map is checking if present inside.